aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/UserPoolConfiguration.kt

```diff
@@ -92,7 +92,9 @@ data class UserPoolConfiguration internal constructor(val builder: Builder) {
                     if (!regex.matches(it))
                         throw Exception("Invalid endpoint")
                 }
-                return endpoint
+                return endpoint?.let {
+                    "https://$endpoint"
```

Comment:
why does https protocol need to be hardcoded here? 
Can't customer provide full url and us have the regex to validate the endpoint?

Reply:
according to the document, customers are not allowed to provide the scheme, path, and params in the url.